flocker/node/agents/blockdevice_manager.py

```diff
@@ -0,0 +1,175 @@
+# -*- test-case-name: flocker.node.agents.test.test_blockdevice_manager -*-
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+This module implements interactions between the OS pertaining to block devices.
+This controls actions such as formatting and mounting a blockdevice.
+"""
+
+import psutil
+from subprocess import CalledProcessError, check_output, STDOUT
+
+from zope.interface import Interface, implementer
+
+from pyrsistent import PClass, field
+
+from twisted.python.filepath import FilePath
+
+from characteristic import attributes
+
+
+@attributes(["blockdevice", "mountpoint", "source_message"])
+class MountError(Exception):
+    """Exception class for errors in attempts to mount."""
```

Comment:
Added ivar documentation.

Changed wording slightly, although importantly if this exception is raised it doesn't 100% guarantee that the mount failed (even by interface spec). This is done because `mount` can exit nonzero even if it successfully mounts a device (if it gets a sigKILL or something). I also don't trust `mkfs` and friends to be idempotent / atomic. A proper implementation of the interface should raise this on any error during the mount process, even if it did succeed, but can't prove that it succeeded.

Potentially a terrible interface, but it'd be sort of hard to do better, and probably not worth the work. At least for this PR I'm just looking to encapsulate the existing interface rather than clean up the interactions.